Aplib.Core/Tactic.cs

```diff
@@ -0,0 +1,167 @@
+                return null;
+
+            if (TacticType == TacticType.Primitive)
+            {
+                PrimitiveTactic tactic = (PrimitiveTactic)this;
+
+                return tactic;
+            }
+
+            return _parent.GetNextTactic();
+        }
+
+        /// <summary>
+        /// Gets the first enabled primitive actions.
+        /// </summary>
+        /// <returns>A list of primitive tactics that are enabled.</returns>
+        public List<PrimitiveTactic> GetFirstEnabledActions()
+        {
+            List<PrimitiveTactic> primitiveTactics = new();
+
+            switch (TacticType)
+            {
+                case TacticType.FirstOf:
+                    foreach (Tactic subTactic in _subTactics)
+                    {
+                        primitiveTactics = subTactic.GetFirstEnabledActions();
+
+                        if (primitiveTactics.Count > 0)
+                            break;
+                    }
+
+                    break;
+                case TacticType.AnyOf:
+                    foreach (Tactic subTactic in _subTactics)
+                    {
+                        primitiveTactics.AddRange(subTactic.GetFirstEnabledActions());
+                    }
+
+                    break;
+                case TacticType.Primitive:
+                    PrimitiveTactic tactic = (PrimitiveTactic)this;
+
+                    if (tactic.IsActionable())
+                        primitiveTactics.Add(tactic);
+
+                    break;
+            }
+
+            return primitiveTactics;
+        }
+
+        /// <summary>
+        /// Determines whether the tactic is actionable.
+        /// </summary>
+        /// <returns>True if the tactic is actionable, false otherwise.</returns>
+        public virtual bool IsActionable() => Guard();
+    }
+
+    /// <summary>
+    /// Represents a primitive tactic in the Aplib.Core namespace.
+    /// </summary>
+    public class PrimitiveTactic : Tactic
+    {
+        /// <summary>
+        /// Gets or sets the action of the primitive tactic.
+        /// </summary>
+        private readonly Action _action;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PrimitiveTactic"/> class with the specified action.
+        /// </summary>
+        /// <param name="action">The action of the primitive tactic.</param>
+        public PrimitiveTactic(Action action) : base(TacticType.Primitive, new()) => _action = action;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PrimitiveTactic"/> class with the specified action.
+        /// </summary>
+        /// <param name="action">The action of the primitive tactic.</param>
+        /// <param name="guard">The guard of the tactic.</param>
+        public PrimitiveTactic(Action action, Func<bool> guard) : base(TacticType.Primitive, new(), guard) => _action = action;
+
+        public override bool IsActionable() => Guard() && _action.IsActionable();
```

Comment:
If the Guard is meant to mimic the base method, consider using `base.IsActionable()` instead:
```suggestion
        public override bool IsActionable() => base.IsActionable() && _action.IsActionable();
```